### PR TITLE
CCDA download problems and unstructured documents

### LINKED ISSUE #5869: CCDA Export download broken
### Describe the bug
Resulting zip of downloads is corrupted and unreliable. Appears to sometimes work. 
@adunsulag Did we need to do a batch download of CCDs for ONC?

I think it may have to do with the action being forwarded for reasons I don't understand(the forward action bit).
Stephen, you noted the forward wasn't needed in comments. Not sure that's true.

A corrupted zip opened with 7_zip as windows explorer won't open.
image

### To Reproduce

Use download option for CCDA export

- Browser:
- OpenEMR version: v7.0.0
- Operating system: Windows

### PR DESCRIPTION
**Pull request by sjpadgett**
- fix issue where download export was generating the ccda twice
- fix issue where last gen'ed document and header was being echoed after zip already created.

Fixes #5869

### CODE REVIEW

**Comment by bradymiller** [comment]
2DV

**Comment by sjpadgett** [comment]
The new patient document exports via an unstructured cda document is complete. Next is the import.
Generally the unstructured document is auto created if the Export Document flag is set and there are documents available in patient chart.
image

The affect is the normal CCDA type is created and then the unstructured doc. Both are then replied back to CCM from ccdaservice where hooks separated the documents and stores in the ccda tabled to be available when creating the download zip. Zip now looks like:
image

I plan next to allow the zip to be uploaded and imported as one or by entire directory. A little more effort for me but worth it to the user IMO. May be able to get to the drag and drop feature already there but broken.

If I may, this takes CCM up another notch and out of the kiddie pool!:)

**Comment by sjpadgett** [comment]
@adunsulag You may be interested to know that our chunking socket send routine is working to 350MB so far.
Although it's really slow rendering and downloading a 150 MB zip. However, I don't think many patients would have documents totaling that.

**Comment by sjpadgett** [comment]
okay now I can start import unstructured.

**Comment by sjpadgett** [comment]
This is a horrible setting max_allowed_packet!!

**Comment by sjpadgett** [comment]
this is weird error in ccm import
An error occurred during execution; please try again later.
Additional information:
Error

File:

    C:\xampp\htdocs\openemr\vendor\laminas\laminas-db\src\TableGateway\AbstractTableGateway.php:495

Message:

    Call to a member function canCallMagicSet() on null

**Comment by sjpadgett** [comment]
Adding to Config to allow disable validation to speed up imports for openemr-openemr transfer

image

File: sql/database.sql [hunk 1]
```diff
@@ -11613,7 +11613,7 @@ CREATE TABLE ccda (
   `uuid` binary(16) DEFAULT NULL,
   pid BIGINT(20) DEFAULT NULL,
   encounter BIGINT(20) DEFAULT NULL,
-  ccda_data MEDIUMTEXT,
+  ccda_data LONGTEXT,
```
Comment (bradymiller) [inline sql/database.sql:11616]: is this change already in the sql upgrade script?

File: interface/modules/zend_modules/module/Carecoordination/src/Carecoordination/Model/EncountermanagerTable.php [hunk 2]
```diff
@@ -332,15 +332,18 @@ public function transmitCcdToRecipients($data = array())
...
+    public function getFileID($pid, $limit = 1)
     {
         $appTable = new ApplicationTable();
         $query = "SELECT cc.id, pd.fname, pd.lname, pd.pid FROM ccda AS cc
 		    LEFT JOIN patient_data AS pd ON pd.pid=cc.pid
 		    WHERE cc.pid = ?
-		    ORDER BY cc.id DESC LIMIT 1";
-        $res = $appTable->zQuery($query, array($pid));
-        $res_cur = $res->current();
+		    ORDER BY cc.id DESC LIMIT ?";
```
[the remark this answers is not in the document]
Comment (sjpadgett) [reply interface/modules/zend_modules/module/Carecoordination/src/Carecoordination/Model/EncountermanagerTable.php:341]: thanks for the catch @bradymiller! Saved me some grief!

**Comment by sjpadgett** [comment]
FYI Virginia ccda
Not importing encounters. Missing a template we expect although not usual to send like this I prob need to allow and fix!
```
<entry typeCode="DRIV">
      <encounter classCode="ENC" moodCode="EVN">
		    <templateId root="2.16.840.1.113883.10.20.22.4.49"/>
        <!-- Encounter activity template -->
        <id root="4800c47a-190d-36c7-0406-218cc2470ebe"/>
        <code  code="162673000" codeSystem="2.16.840.1.113883.6.96" displayName="General examination of patient (procedure)">
          <originalText><reference value="#encounters-desc-2"/></originalText>
        </code>
        <text>
          <reference value="#encounters-desc-2"/>
        </text>
        <effectiveTime>
          <low value="20111110071615"/>
          <high value="20111110073115"/>
        </effectiveTime>
      </encounter>
    </entry>
```

**Comment by sjpadgett** [comment]
Here we go:

image


Cool!

**Pull request merged**